### ISSUE TITLE
Golang events are not parsed at all. 

### ISSUE THREAD

**Issue by coinsandsteeldev** [loomnetwork/unity3d-sdk#45]
This is Contract.cs method which is used to pre-process event before emitting it to listeners.
```
protected override ChainEventArgs TransformChainEvent(RawChainEventArgs e) {
    string jsonRpcEventString = Encoding.UTF8.GetString(e.Data);
    JsonRpcEvent jsonRpcEvent = JsonConvert.DeserializeObject<JsonRpcEvent>(jsonRpcEventString);
    byte[] eventData = Encoding.UTF8.GetBytes(jsonRpcEvent.Data);

    return new ChainEventArgs(
         e.ContractAddress,
         e.CallerAddress,
         e.BlockHeight,
         eventData,
         jsonRpcEvent.Method
    );
}
```
When i emit event in Go either using `ctx.Emit()` or `ctx.EmitTopics()` i have same result:

`jsonRpcEventString ` has serialized json from Emit for instance

```
type AccountCreatedEvt struct {
	Test  string
	Test2 int
}
```

Will result as `{"Test": "test", "Test2": 1488}`. So the next line fails to deserialize
`JsonRpcEvent jsonRpcEvent = JsonConvert.DeserializeObject<JsonRpcEvent>(jsonRpcEventString);`

Resulting in converting null to byte[]

I wonder whether it is problem of Loom Node emitting event in the wrong format or something else.

**Comment by coinsandsteeldev** [comment]
I'm constrained by this issue a lot. And i think this is huge issue right now.

**Comment by coinsandsteeldev** [comment]
It seems like we can emit anything we want in Go contract. And i think we need better approach to TransformChainEvent, i would make it injection, with some standard one by default. Right now if i want to have pure protobuf events i need to open source code of SDK and change it. So i would propose this simple change which will make life easier for future generations :)

**Comment by ZimM-LostPolygon** [comment]
@coinsandsteeldev Not exactly sure I understand the issue. The first argument of `ctx.Emit()` or `ctx.EmitTopics()` goes into `JsonRpcEvent.Data`. And sure, you can put pretty much anything there. But you say that it replaces the complete `RawChainEventArgs.Data`?..

**Comment by coinsandsteeldev** [comment]
@ZimM-LostPolygon No it doesn't. It fail to parse JsonRpcEvent at all.

**Comment by coinsandsteeldev** [comment]
Of course if you follow JsonRpcEvent layout it will be ok, but i don't think it is a way to go. It is fixed easily, and it is done in my own version of your SDK.

**Comment by ZimM-LostPolygon** [comment]
Yep, I can confirm there is an issue. Meanwhile, you probably should subclass `Contract` and override `TransformChainEvent`.
